src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1667,6 +1693,69 @@ protected override void HandleModulesRequestAsync(IRequestResponder<ModulesArgum
             responder.SetResponse(response);
         }
 
+        // test -- need to delete
+        protected override void HandleDisassembleRequestAsync(IRequestResponder<DisassembleArguments, DisassembleResponse> responder)
+        {
+            DisassembleResponse response = new DisassembleResponse();
+            DisassembleArguments disassembleArguments = responder.Arguments;
+            Debug.Assert(!string.IsNullOrEmpty(disassembleArguments.MemoryReference));
+            try
+            {
+                if (GetMemoryContext(disassembleArguments.MemoryReference, disassembleArguments.Offset, out IDebugMemoryContext2 memoryContext, out ulong address) == HRConstants.S_OK)
```

Comment:
>if (GetMemoryContext(disassembleArguments.MemoryReference, disassembleArguments.Offset, out IDebugMemoryContext2 memoryContext, out ulong address) == HRConstants.S_OK) [](start = 16, length = 167)

Rather than adding a different resource string for each failure and ignoring the HRESULTs that you get, I would instead suggest defining one new resource string: `Error_Scenario_Disassemble = "Unable to disassemble. {0}"` and you can do:

```C#
            var eb = new ErrorBuilder(() => AD7Resources.Error_Scenario_Disassemble);

            eb.CheckHR(GetMemoryContext(disassembleArguments.MemoryReference, disassembleArguments.Offset, out IDebugMemoryContext2 memoryContext, out ulong address));
            
            IDebugCodeContext2 codeContext = memoryContext as IDebugCodeContext2;
            if (codeContext == null)
            {
                eb.CheckHR(HRConstants.E_NOTIMPL);
            }

            eb.CheckOutput(m_program.GetDisassemblyStream(enum_DISASSEMBLY_STREAM_SCOPE.DSS_ALL, codeContext, out IDebugDisassemblyStream2 disassemblyStream));
```